packages/tour-kit/src/components/tour-kit-frame.tsx

```diff
@@ -183,13 +183,29 @@ const TourKitFrame: React.FunctionComponent< Props > = ( { config } ) => {
 		}
 	}, [ popperUpdate, referenceElement ] );
 
-	const classNames = classnames(
+	const classSplit = ( classes: string | Array< string > | undefined ) => {
+		if ( typeof classes === 'string' ) {
+			return classes.split( ',' );
+		}
+		return classes;
+	};
+
+	const frameClasses = classnames(
 		'tour-kit-frame',
-		config.options?.className,
+		classSplit( config.options?.className ),
 		isMobile ? 'is-mobile' : 'is-desktop',
 		{ 'is-visible': tourReady }
 	);
 
+	const stepClasses = classnames(
+		'tour-kit-step',
+		`is-step-${ currentStepIndex }`,
+		config.options?.className
+			? classSplit( config.options?.className )?.map( ( value ) => value + '_step' )
```

Comment:
Should be `__step`.

However, I'm having second thoughts about the inclusion of these "magic" classes. They sort of lost their appeal to me, especially now that we handle multiple classes from the root component. Imagine a scenario where we pass `foo-tour` and `is-foo-tour-modifier` to the parent, we end up with `foo-tour__step` and `is-foo-tour-modifier__step` in the step. Feels complex, hard to communicate, and hard to be trusted by the consumer. I would vote to remove this part altogether.

What do you think? If removing, we can just send [wpcom-editor-welcome-tour__step](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/style-tour.scss#L220) along with `is-with-extra-padding` now that we handle multiple classes for steps.

Reply:
Cool, I probably should have raised this concern when I put this together. I agree the classes get a bit excessive when we add `__step` to all the parent classes. I will clean this up 🕺

Reply:
Cool! Thanks, and please feel free to raise concerns at any point. 😃